llvm/include/llvm/Passes/MachinePassRegistry.def

```diff
@@ -125,8 +125,8 @@ MACHINE_FUNCTION_ANALYSIS("pass-instrumentation", PassInstrumentationAnalysis,
 #ifndef MACHINE_FUNCTION_PASS
 #define MACHINE_FUNCTION_PASS(NAME, PASS_NAME, CONSTRUCTOR)
 #endif
+MACHINE_FUNCTION_PASS("free-machine-function", FreeMachineFunctionPass, ())
```

Comment:
do we somehow block running this with -run-pass? What happens if you manually run this on its own or before other passes?